### ISSUE TITLE
nvidia-docker does not seem to work with nvidia-driver ver 381.09

### ISSUE THREAD

**Issue by honglh** [NVIDIA/nvidia-docker#427]
My nvidia machine was using driver 375.39. After the driver is updated to 381.09, the nvidia-docker cannot start anymore (various errors). 

Before I elaborate on errors, can someone tell if this is expected?

**Comment by flx42** [comment]
No, it's not expected.
But you should also upgrade to 381.22

**Comment by redmud** [comment]
which step populates /var/lib/nvidia-docker/volumes?
Here I see /var/lib/nvidia-docker/volumes/375.39 where it has
/var/lib/nvidia-docker/volumes/375.3/lib64

But there is no volume for 381.09.

Reinstalling nvidia-docker does not make any difference.  How can I result in a volume for 381.09?

**Comment by flx42** [comment]
Restarting nvidia-docker after the driver install (or rebooting the machine) should be sufficient.
What's the output of `nvidia-smi`?
What's the output of `ldconfig -p | grep -E 'libcuda|libnvidia'`?

**Comment by redmud** [comment]
```[root@labhpc1-gpu2 var]# nvidia-smi 
Wed Jul 12 11:20:57 2017       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 381.09                 Driver Version: 381.09                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX TIT...  Off  | 0000:02:00.0     Off |                  N/A |
| 22%   42C    P8    18W / 250W |    513MiB / 12207MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce GTX TIT...  Off  | 0000:03:00.0     Off |                  N/A |
| 22%   48C    P8    18W / 250W |    512MiB / 12207MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  GeForce GTX TIT...  Off  | 0000:84:00.0     Off |                  N/A |
| 22%   44C    P8    17W / 250W |    512MiB / 12207MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  GeForce GTX TIT...  Off  | 0000:85:00.0     Off |                  N/A |
| 22%   46C    P8    17W / 250W |    512MiB / 12207MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID  Type  Process name                               Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
[root@labhpc1-gpu2 var]# ldconfig -p | grep -E 'libcuda|libnvidia'
	libnvidia-tls.so.381.09 (libc6,x86-64, hwcap: 0x8000000000000000, OS ABI: Linux 2.3.99) => /usr/lib64/tls/libnvidia-tls.so.381.09
	libnvidia-tls.so.381.09 (libc6,x86-64, OS ABI: Linux 2.2.5) => /usr/lib64/libnvidia-tls.so.381.09
	libnvidia-tls.so.381.09 (libc6, OS ABI: Linux 2.2.5) => /lib/libnvidia-tls.so.381.09
	libnvidia-ptxjitcompiler.so.381.09 (libc6,x86-64) => /usr/lib64/libnvidia-ptxjitcompiler.so.381.09
	libnvidia-ptxjitcompiler.so.381.09 (libc6) => /lib/libnvidia-ptxjitcompiler.so.381.09
	libnvidia-opencl.so.1 (libc6,x86-64) => /usr/lib64/libnvidia-opencl.so.1
	libnvidia-opencl.so.1 (libc6) => /lib/libnvidia-opencl.so.1
	libnvidia-ml.so.1 (libc6,x86-64) => /usr/lib64/libnvidia-ml.so.1
	libnvidia-ml.so.1 (libc6) => /lib/libnvidia-ml.so.1
	libnvidia-ml.so (libc6,x86-64) => /usr/lib64/libnvidia-ml.so
	libnvidia-ml.so (libc6) => /lib/libnvidia-ml.so
	libnvidia-ifr.so.1 (libc6,x86-64) => /usr/lib64/libnvidia-ifr.so.1
	libnvidia-ifr.so.1 (libc6) => /lib/libnvidia-ifr.so.1
	libnvidia-ifr.so (libc6,x86-64) => /usr/lib64/libnvidia-ifr.so
	libnvidia-ifr.so (libc6) => /lib/libnvidia-ifr.so
	libnvidia-gtk3.so.381.09 (libc6,x86-64) => /usr/lib64/libnvidia-gtk3.so.381.09
	libnvidia-gtk2.so.381.09 (libc6,x86-64) => /usr/lib64/libnvidia-gtk2.so.381.09
	libnvidia-glsi.so.381.09 (libc6,x86-64) => /usr/lib64/libnvidia-glsi.so.381.09
	libnvidia-glsi.so.381.09 (libc6) => /lib/libnvidia-glsi.so.381.09
	libnvidia-glcore.so.381.09 (libc6,x86-64) => /usr/lib64/libnvidia-glcore.so.381.09
	libnvidia-glcore.so.381.09 (libc6) => /lib/libnvidia-glcore.so.381.09
	libnvidia-fbc.so.1 (libc6,x86-64) => /usr/lib64/libnvidia-fbc.so.1
	libnvidia-fbc.so.1 (libc6) => /lib/libnvidia-fbc.so.1
	libnvidia-fbc.so (libc6,x86-64) => /usr/lib64/libnvidia-fbc.so
	libnvidia-fbc.so (libc6) => /lib/libnvidia-fbc.so
	libnvidia-fatbinaryloader.so.381.09 (libc6,x86-64) => /usr/lib64/libnvidia-fatbinaryloader.so.381.09
	libnvidia-fatbinaryloader.so.381.09 (libc6) => /lib/libnvidia-fatbinaryloader.so.381.09
	libnvidia-encode.so.1 (libc6,x86-64) => /usr/lib64/libnvidia-encode.so.1
	libnvidia-encode.so.1 (libc6) => /lib/libnvidia-encode.so.1
	libnvidia-encode.so (libc6,x86-64) => /usr/lib64/libnvidia-encode.so
	libnvidia-encode.so (libc6) => /lib/libnvidia-encode.so
	libnvidia-eglcore.so.381.09 (libc6,x86-64) => /usr/lib64/libnvidia-eglcore.so.381.09
	libnvidia-eglcore.so.381.09 (libc6) => /lib/libnvidia-eglcore.so.381.09
	libnvidia-egl-wayland.so.1.0.1 (libc6,x86-64) => /usr/lib64/libnvidia-egl-wayland.so.1.0.1
	libnvidia-compiler.so.381.09 (libc6,x86-64) => /usr/lib64/libnvidia-compiler.so.381.09
	libnvidia-compiler.so.381.09 (libc6) => /lib/libnvidia-compiler.so.381.09
	libnvidia-cfg.so.1 (libc6,x86-64) => /usr/lib64/libnvidia-cfg.so.1
	libnvidia-cfg.so (libc6,x86-64) => /usr/lib64/libnvidia-cfg.so
	libcudart.so.8.0 (libc6,x86-64) => /usr/local/cuda/lib64/libcudart.so.8.0
	libcudart.so (libc6,x86-64) => /usr/local/cuda/lib64/libcudart.so
	libcuda.so.1 (libc6,x86-64) => /usr/lib64/libcuda.so.1
	libcuda.so.1 (libc6) => /lib/libcuda.so.1
	libcuda.so (libc6,x86-64) => /usr/lib64/libcuda.so
	libcuda.so (libc6) => /lib/libcuda.so
```

**Comment by flx42** [comment]
And what's the error when you try to use nvidia-docker?

**Comment by redmud** [comment]
If I run this command
```nvidia-docker run --rm nvidia/cuda nvidia-smi```
with 
```/var/lib/nvidia-docker/volumes/nvidia_driver/375.39```
and try to run a comtainer
```nvidia-docker run --rm nvidia/cuda nvidia-smi```
I would get this error
```NVIDIA-SMI couldn't find libnvidia-ml.so library in your system. Please make sure that the NVIDIA Display Driver is properly installed and present in your system.
```
which may be expected given the volume 375.39 does not have 381.09 libraries
```
ls -l /var/lib/nvidia-docker/volumes/nvidia_driver/375.39/li64
lrwxrwxrwx. 1  992  989       22 Apr 26 04:25 libnvidia-ml.so.1 -> libnvidia-ml.so.375.39
-rwxr-xr-x. 1 root root  1197664 Feb 16 09:48 libnvidia-ml.so.375.39
```

**Comment by flx42** [comment]
> which may be expected given the volume 375.39 does not have 381.09 libraries

They should be in a separate folder.
I'm not sure what's happening with your setup, what's the output of:
```
journalctl -n -u nvidia-docker
```
And
```
curl -s localhost:3476/docker/cli
```
Also try to restart the nvidia-docker plugin too `sudo systemctl restart nvidia-docker`.

**Comment by flx42** [comment]
@redmud: did you solve your issue?

**Comment by callemein** [comment]
I'm facing the same error as @redmud. After updating to driver 384.90 which works fine on my desktop, I'm getting an error with `nvidia-docker`. 

I've tried all the steps as explained above, with no success.
I'll post my output of the above suggestions here:

`$ nvidia-smi `
	```
Mon Oct  2 09:59:02 2017       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 384.90                 Driver Version: 384.90                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 108...  Off  | 00000000:03:00.0 Off |                  N/A |
|  0%   36C    P8    11W / 280W |      1MiB / 11172MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Quadro K5000        Off  | 00000000:22:00.0  On |                  Off |
| 30%   42C    P8    19W / 137W |    204MiB /  4034MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    1      2976      G   /usr/lib/xorg/Xorg                           202MiB |
+-----------------------------------------------------------------------------+
	```

`$ ldconfig -p | grep -E 'libcuda|libnvidia'`
	```
libnvidia-wfb.so.1 (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-wfb.so.1
	libnvidia-wfb.so.1 (libc6) => /usr/lib32/nvidia-384/libnvidia-wfb.so.1
	libnvidia-tls.so.384.90 (libc6,x86-64, hwcap: 0x8000000000000000, OS ABI: Linux 2.3.99) => /usr/lib/nvidia-384/tls/libnvidia-tls.so.384.90
	libnvidia-tls.so.384.90 (libc6,x86-64, OS ABI: Linux 2.2.5) => /usr/lib/nvidia-384/libnvidia-tls.so.384.90
	libnvidia-tls.so.384.90 (libc6, hwcap: 0x8000000000000000, OS ABI: Linux 2.3.99) => /usr/lib32/nvidia-384/tls/libnvidia-tls.so.384.90
	libnvidia-tls.so.384.90 (libc6, OS ABI: Linux 2.2.5) => /usr/lib32/nvidia-384/libnvidia-tls.so.384.90
	libnvidia-ptxjitcompiler.so.1 (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-ptxjitcompiler.so.1
	libnvidia-ptxjitcompiler.so.1 (libc6) => /usr/lib32/nvidia-384/libnvidia-ptxjitcompiler.so.1
	libnvidia-ml.so.1 (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-ml.so.1
	libnvidia-ml.so.1 (libc6) => /usr/lib32/nvidia-384/libnvidia-ml.so.1
	libnvidia-ml.so (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-ml.so
	libnvidia-ml.so (libc6) => /usr/lib32/nvidia-384/libnvidia-ml.so
	libnvidia-ifr.so.1 (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-ifr.so.1
	libnvidia-ifr.so.1 (libc6) => /usr/lib32/nvidia-384/libnvidia-ifr.so.1
	libnvidia-ifr.so (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-ifr.so
	libnvidia-ifr.so (libc6) => /usr/lib32/nvidia-384/libnvidia-ifr.so
	libnvidia-gtk3.so.384.90 (libc6,x86-64) => /usr/lib/libnvidia-gtk3.so.384.90
	libnvidia-gtk2.so.384.90 (libc6,x86-64) => /usr/lib/libnvidia-gtk2.so.384.90
	libnvidia-glsi.so.384.90 (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-glsi.so.384.90
	libnvidia-glsi.so.384.90 (libc6) => /usr/lib32/nvidia-384/libnvidia-glsi.so.384.90
	libnvidia-glcore.so.384.90 (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-glcore.so.384.90
	libnvidia-glcore.so.384.90 (libc6) => /usr/lib32/nvidia-384/libnvidia-glcore.so.384.90
	libnvidia-fbc.so.1 (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-fbc.so.1
	libnvidia-fbc.so.1 (libc6) => /usr/lib32/nvidia-384/libnvidia-fbc.so.1
	libnvidia-fbc.so (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-fbc.so
	libnvidia-fbc.so (libc6) => /usr/lib32/nvidia-384/libnvidia-fbc.so
	libnvidia-fatbinaryloader.so.384.90 (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-fatbinaryloader.so.384.90
	libnvidia-fatbinaryloader.so.384.90 (libc6) => /usr/lib32/nvidia-384/libnvidia-fatbinaryloader.so.384.90
	libnvidia-encode.so.1 (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-encode.so.1
	libnvidia-encode.so.1 (libc6) => /usr/lib32/nvidia-384/libnvidia-encode.so.1
	libnvidia-encode.so (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-encode.so
	libnvidia-encode.so (libc6) => /usr/lib32/nvidia-384/libnvidia-encode.so
	libnvidia-eglcore.so.384.90 (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-eglcore.so.384.90
	libnvidia-eglcore.so.384.90 (libc6) => /usr/lib32/nvidia-384/libnvidia-eglcore.so.384.90
	libnvidia-egl-wayland.so.1.0.1 (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-egl-wayland.so.1.0.1
	libnvidia-egl-wayland.so.1.0.1 (libc6) => /usr/lib32/nvidia-384/libnvidia-egl-wayland.so.1.0.1
	libnvidia-compiler.so.384.90 (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-compiler.so.384.90
	libnvidia-compiler.so.384.90 (libc6) => /usr/lib32/nvidia-384/libnvidia-compiler.so.384.90
	libnvidia-compiler.so (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-compiler.so
	libnvidia-compiler.so (libc6) => /usr/lib32/nvidia-384/libnvidia-compiler.so
	libnvidia-cfg.so.1 (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-cfg.so.1
	libnvidia-cfg.so.1 (libc6) => /usr/lib32/nvidia-384/libnvidia-cfg.so.1
	libnvidia-cfg.so (libc6,x86-64) => /usr/lib/nvidia-384/libnvidia-cfg.so
	libnvidia-cfg.so (libc6) => /usr/lib32/nvidia-384/libnvidia-cfg.so

```

Clean boot output of `journalctl -n -u nvidia-docker`
```

-- Logs begin at Mon 2017-10-02 09:41:40 CEST, end at Mon 2017-10-02 10:03:22 CEST. --
Oct 02 10:03:21 DELL-TCA nvidia-docker-plugin[13584]: /usr/bin/nvidia-docker-plugin | 2017/10/02 10:03:21 Discovering GPU devices
Oct 02 10:03:21 DELL-TCA systemd[1]: Started NVIDIA Docker plugin.
Oct 02 10:03:21 DELL-TCA nvidia-docker-plugin[13584]: /usr/bin/nvidia-docker-plugin | 2017/10/02 10:03:21 Error: cuda: CUDA driver version is insufficient for CUDA runtime version
Oct 02 10:03:21 DELL-TCA systemd[1]: nvidia-docker.service: Main process exited, code=exited, status=1/FAILURE
Oct 02 10:03:21 DELL-TCA systemd[1]: nvidia-docker.service: Unit entered failed state.
Oct 02 10:03:21 DELL-TCA systemd[1]: nvidia-docker.service: Failed with result 'exit-code'.
Oct 02 10:03:22 DELL-TCA systemd[1]: nvidia-docker.service: Service hold-off time over, scheduling restart.
Oct 02 10:03:22 DELL-TCA systemd[1]: Stopped NVIDIA Docker plugin.
Oct 02 10:03:22 DELL-TCA systemd[1]: nvidia-docker.service: Start request repeated too quickly.
Oct 02 10:03:22 DELL-TCA systemd[1]: Failed to start NVIDIA Docker plugin.

```

**Comment by 3XX0** [comment]
Closing as it most likely a driver installation issue and/or fixed in 2.0/master.
